materialize/resources/resource_connection.go

```diff
@@ -565,36 +565,50 @@ func (b *ConnectionBuilder) Create() string {
 			if b.kafkaSSHTunnel != "" {
 				q.WriteString(`BROKERS (`)
 				for i, broker := range b.kafkaBrokers {
-					q.WriteString(fmt.Sprintf(`'%s' USING SSH TUNNEL %s`, broker, b.kafkaSSHTunnel))
+					q.WriteString(fmt.Sprintf(`'%s' USING SSH TUNNEL %s`, broker["broker"], b.kafkaSSHTunnel))
 					if i < len(b.kafkaBrokers)-1 {
 						q.WriteString(`,`)
 					}
 				}
 				q.WriteString(`)`)
 			} else {
-				q.WriteString(fmt.Sprintf(`BROKERS ('%s')`, strings.Join(b.kafkaBrokers, "', '")))
+				q.WriteString(fmt.Sprintf(`BROKERS (`))
+				for i, broker := range b.kafkaBrokers {
```

Comment:
Seems like you didn't have the problem but I was running into issues with the tests because `schema.TypeMap` aren't ordered. I had to do a couple other steps:

https://github.com/MaterializeInc/terraform-provider-materialize/blob/main/materialize/resources/resource_source.go#L344-L357

But maybe not an issue?

Reply:
I've just had another look, so it looks like that the resource is getting prepared as expected:
```
  + resource "materialize_connection" "example_kafka_connection" {
      + connection_type       = "KAFKA"
      + database_name         = "materialize"
      + id                    = (known after apply)
      + kafka_brokers         = [
          + {
              + "availability_zone"      = "use1-az1"
              + "broker"                 = "b-1.hostname-1:9096"
              + "privatelink_connection" = "privatelink_conn"
              + "target_group_port"      = "9001"
            },
          + {
              + "availability_zone"      = "use1-az2"
              + "broker"                 = "b-1.hostname-1:9097"
              + "privatelink_connection" = "privatelink_conn"
              + "target_group_port"      = "9002"
            },
        ]
      + kafka_progress_topic  = "example"
      + kafka_sasl_mechanisms = "SCRAM-SHA-256"
      + kafka_sasl_password   = "kafka_password"
      + kafka_sasl_username   = "example"
      + name                  = "example_kafka_connection"
      + schema_name           = "public"
    }
  ```
  
  And also the tests produce the correct SQL, but if I try to apply it I hit this problem:
  
  ```
  Stack trace from the terraform-provider-materialize plugin:

panic: interface conversion: interface {} is []interface {}, not []map[string]interface {}

goroutine 83 [running]:
terraform-materialize/materialize/resources.connectionCreate({0x1050b4fa8, 0x140001a5140}, 0x0?, {0x1050a6400?, 0x14000589200?})
       terraform-materialize/terraform-provider-materialize/materialize/resources/resource_connection.go:710 +0xbec
```

Reply:
Managed to fix that problem!